great_expectations/checkpoint/util.py

```diff
@@ -266,25 +267,18 @@ def get_substituted_batch_request(
         batch_request=substituted_runtime_batch_request
     )
 
-    effective_batch_request: dict = dict(
-        **substituted_runtime_batch_request, **validation_batch_request
-    )
-
-    batch_request_class: type
-
-    if "runtime_parameters" in effective_batch_request:
-        batch_request_class = RuntimeBatchRequest
-    else:
-        batch_request_class = BatchRequest
-
     for key, value in validation_batch_request.items():
         substituted_value = substituted_runtime_batch_request.get(key)
         if value is not None and substituted_value is not None:
             raise ge_exceptions.CheckpointError(
                 f'BatchRequest attribute "{key}" was specified in both validation and top-level CheckpointConfig.'
             )
 
-    return batch_request_class(**effective_batch_request)
+    effective_batch_request: dict = dict(
+        **substituted_runtime_batch_request, **validation_batch_request
+    )
+
+    return materialize_batch_request(batch_request=effective_batch_request)
```

Comment:
Double checking : does this the `get_substituted_batch_request()` function still return `Optional[Union[BatchRequest, RuntimeBatchRequest]]`? Or is it `Union[BatchRequest, RuntimeBatchRequest]`

Reply:
@Shinnnyshinshin Good catch -- fixed!  Thanks!